### DISCUSSION TITLE
command+enter on textarea

### DISCUSSION THREAD

**Discussion by tordans** [react-hook-form/react-hook-form#9485] [Q&A] [answered]
Hey, I have this litte app with a textarea that should be submitted on <kbd>command+enter</kbd>.

* The form https://github.com/osmberlin/osm-tag-updater/blob/main/src/pages/PageHome/InputTags/InputTags.tsx
* The app https://osmberlin.github.io/osm-tag-updater/

I tried applying https://www.peterbe.com/plog/command-enter-to-submit-form-focus-textarea-react with a shared ref https://react-hook-form.com/faqs#Howtosharerefusage, but the keyboard commands did not register (or something).

Anyone have an example of this somewhere?

**Comment by leapful** [comment]
@tordans 

Please provide the codesandbox url to re-produce the issue. Based on your repo, there is no keyboard event handling.

**Reply by tordans** [reply]
Thanks @leapful for the implicit nudge "that should work". I needed this to give it another go (or to provide a proper broken version). Turns out, this is how it works: https://github.com/osmberlin/osm-tag-updater/blob/main/src/pages/PageHome/InputTags/InputTags.tsx

In case you are interested in some details:
1. For some reason the submit only worked like I have it here, not the way I found in the examples, see comment at https://github.com/osmberlin/osm-tag-updater/blob/main/src/pages/PageHome/InputTags/InputTags.tsx#L28-L30
2. And another feedback on https://react-hook-form.com/faqs#Howtosharerefusage: I had a bug because I did not notice the `name=firstName` which appeared after I copied the example. Its in `input {...rest} name="firstName" ref`. I think this example would be better without the explicit name, since that is provided by the `{...rest}` part.

**Reply by leapful** [reply]
@tordans :

In case you want to simplify the implementation, just use form element to dispatch submit event.

Working demo: https://codesandbox.io/s/gracious-davinci-7fcmj5?file=/src/App.js

**Comment by Phibee** [comment, marked as answer]
@tordans , bro you can just directly add onKeyDown event on your text area and create callback function like handleKeyDown.

<img width="1349" alt="image" src="https://user-images.githubusercontent.com/3889714/205022597-27c983d7-a1a1-4181-8996-f20d62efaa2f.png">

**Reply by tordans** [reply]
Thanks @Phibee and @leapful for the help, this is so much cleaner now! https://github.com/osmberlin/osm-tag-updater/commit/dcc3d0886875a8c1a39d5fe84a69aa5e89c4a90b
